builder/azure/arm/step_capture_image.go

```diff
@@ -44,89 +45,90 @@ func NewStepCaptureImage(client *AzureClient, ui packersdk.Ui) *StepCaptureImage
 	var isSIGImage = state.Get(constants.ArmIsSIGImage).(bool)
 	var skipGeneralization = state.Get(constants.ArmSharedImageGalleryDestinationSpecialized).(bool)
+
+	vmId := hashiVMSDK.NewVirtualMachineID(subscriptionId, resourceGroupName, computeName)
 	s.say(fmt.Sprintf(" -> Compute ResourceGroupName : '%s'", resourceGroupName))
 	s.say(fmt.Sprintf(" -> Compute Name              : '%s'", computeName))
 	s.say(fmt.Sprintf(" -> Compute Location          : '%s'", location))
 
-	var err error
 	if skipGeneralization {
 		s.say("Skipping generalization of Compute Gallery Image")
 	} else {
 		s.say("Generalizing machine ...")
-		err = s.generalizeVM(resourceGroupName, computeName)
-	}
+		err := s.generalizeVM(vmId)
+
+		if err == nil {
+			if isManagedImage {
+				s.say("Capturing image ...")
+				var targetManagedImageResourceGroupName = state.Get(constants.ArmManagedImageResourceGroupName).(string)
+				var targetManagedImageName = state.Get(constants.ArmManagedImageName).(string)
+				var targetManagedImageLocation = state.Get(constants.ArmLocation).(string)
+				s.say(fmt.Sprintf(" -> Image ResourceGroupName   : '%s'", targetManagedImageResourceGroupName))
+				s.say(fmt.Sprintf(" -> Image Name                : '%s'", targetManagedImageName))
+				s.say(fmt.Sprintf(" -> Image Location            : '%s'", targetManagedImageLocation))
+				err = s.captureManagedImage(ctx, subscriptionId, targetManagedImageResourceGroupName, targetManagedImageName, imageParameters)
+			} else if isSIGImage {
+				// It's possible to create SIG image
+				return multistep.ActionContinue
+			} else {
+				s.say("Capturing VHD ...")
+				err = s.captureVhd(ctx, vmId, vmCaptureParameters)
+			}
+		}
+		if err != nil {
+			state.Put(constants.Error, err)
+			s.error(err)
 
-	if err == nil {
-		if isManagedImage {
-			s.say("Capturing image ...")
-			var targetManagedImageResourceGroupName = state.Get(constants.ArmManagedImageResourceGroupName).(string)
-			var targetManagedImageName = state.Get(constants.ArmManagedImageName).(string)
-			var targetManagedImageLocation = state.Get(constants.ArmLocation).(string)
-			s.say(fmt.Sprintf(" -> Image ResourceGroupName   : '%s'", targetManagedImageResourceGroupName))
-			s.say(fmt.Sprintf(" -> Image Name                : '%s'", targetManagedImageName))
-			s.say(fmt.Sprintf(" -> Image Location            : '%s'", targetManagedImageLocation))
-			err = s.captureManagedImage(ctx, targetManagedImageResourceGroupName, targetManagedImageName, imageParameters)
-		} else if isSIGImage {
-			// It's possible to create SIG image
-			return multistep.ActionContinue
-		} else {
-			s.say("Capturing VHD ...")
-			err = s.captureVhd(ctx, resourceGroupName, computeName, vmCaptureParameters)
+			return multistep.ActionHalt
 		}
-	}
-	if err != nil {
-		state.Put(constants.Error, err)
-		s.error(err)
 
-		return multistep.ActionHalt
+		// HACK(chrboum): I do not like this.  The capture method should be returning this value
```

Comment:
Who's `cherboum` and `paulmey`? Are they working on the SDK?
Oh nevermind that's an older comment that was moved, still, is this hack still relevant with the refactor?

Reply:
Yes but I haven't fully vetted that, there are two helper methods we use in azure client that still rely on autorest specific libraries, these are working on the new SDK currently but I need to migrate them still